scala/private/phases/phase_compile.bzl

```diff
@@ -165,6 +165,7 @@ def _phase_compile(
         class_jar = out.class_jar,
         coverage = out.coverage.external,
         full_jars = out.full_jars,
+        files = depset(out.full_jars),
```

Comment:
having a depset here is probably not the best idea since depsets have their cost and especially since you want to customize this in other phases then you might have depsets which are nested for no reason.
Maybe pass array?

Reply:
I'm not sure. In some case we may wind up passing along existing depsets, but in many others we have just a few files that need to be added in. It's not clear to me which is more common and the impact to analysis time.

Reply:
Are you replying in general or about files and runfiles? What we have now in these cases are arrays. If you’ll want to override these then you’ll need to have a nested depset.
In general I think it will depend on the case.